appinventor/components/src/com/google/appinventor/components/runtime/util/ViewUtil.java

```diff
@@ -189,9 +199,57 @@ public static void setBackgroundImage(View view, Drawable drawable) {
   /**
    * Sets the image for an ImageView.
    */
-  public static void setImage(ImageView view, Drawable drawable) {
-    view.setImageDrawable(drawable);
-    if (drawable != null) {
+  public static void setImage(ComponentContainer container, ImageView view, String picturePath) {
+    if (picturePath.endsWith(".gif") && Build.VERSION.SDK_INT >= Build.VERSION_CODES.GINGERBREAD_MR1) {
+      GlideDrawableImageViewTarget imageViewTarget = new GlideDrawableImageViewTarget(view);
+      MediaUtil.MediaSource mediaSource = MediaUtil.determineMediaSource(container.$form(), picturePath);
+      switch (mediaSource) {
+        case ASSET:
+          try {
+            InputStream is = container.$form().getAssets().open(picturePath);
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            byte[] buf = new byte[4096];
+            int read;
+            while ((read = is.read(buf)) > 0) {
+              bos.write(buf, 0, read);
+            }
+            buf = bos.toByteArray();
+            Glide.with(container.$context())
+                .load(buf)
+                .into(imageViewTarget);
+          } catch (IOException e) {
+            Log.e("Image", "Unable to load " + picturePath);
```

Comment:
We may want to create a new error code and report this via the Form in case the developer wants to load a fallback image.